### PR TITLE
Introduce ChannelPriceHistoryConfig

### PR DESCRIPTION
**Pull request by coldic3**

### CODE REVIEW

File: src/Infrastructure/Migrations/Version20230321102453.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        $this->addSql('ALTER TABLE sylius_channel_price_history_config_excluded_taxons ADD CONSTRAINT FK_77FD02ADE13F470 FOREIGN KEY (taxon_id) REFERENCES sylius_taxon (id) ON DELETE CASCADE');
+        $this->addSql('ALTER TABLE sylius_channel_excluded_taxons DROP FOREIGN KEY FK_C5BAE23572F5A1AA');
+        $this->addSql('ALTER TABLE sylius_channel_excluded_taxons DROP FOREIGN KEY FK_C5BAE235DE13F470');
+        $this->addSql('DROP TABLE sylius_channel_excluded_taxons');
+        $this->addSql('ALTER TABLE sylius_channel ADD channel_price_history_config_id INT DEFAULT NULL, DROP lowest_price_for_discounted_products_visible, DROP lowest_price_for_discounted_products_checking_period');
+        $this->addSql('ALTER TABLE sylius_channel ADD CONSTRAINT FK_16C8119E75F20EAE FOREIGN KEY (channel_price_history_config_id) REFERENCES sylius_channel_price_history_config (id) ON DELETE CASCADE');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_16C8119E75F20EAE ON sylius_channel (channel_price_history_config_id)');
+        $this->addSql('ALTER TABLE sylius_channel_pricing_log_entry RENAME INDEX idx_b3f5aac23eadffe5 TO IDX_77181A53EADFFE5');
```
Comment (GSadee) [inline src/Infrastructure/Migrations/Version20230321102453.php:40]: While we do not need to migrate the current data, we should probably add a migration with the addition of a basic configuration for each channel

File: src/Infrastructure/Serializer/ChannelDenormalizer.php [hunk 2]
```diff
@@ -13,21 +13,21 @@
...
 use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
 use Webmozart\Assert\Assert;
 
-final class ChannelExcludedTaxonsDenormalizer implements ContextAwareDenormalizerInterface, DenormalizerAwareInterface
+final class ChannelDenormalizer implements ContextAwareDenormalizerInterface, DenormalizerAwareInterface
 {
     use DenormalizerAwareTrait;
 
-    private const ALREADY_CALLED = 'sylius_channel_excluded_taxons_denormalizer_already_called';
+    private const ALREADY_CALLED = 'sylius_price_history_channel_denormalizer_already_called';
```
Comment (GSadee) [inline src/Infrastructure/Serializer/ChannelDenormalizer.php:28]:
```suggestion
    private const ALREADY_CALLED = 'sylius_channel_denormalizer_already_called';
```
?

File: src/Infrastructure/Migrations/Version20230321102453.php (hunk 1 above)
Comment (NoResponseMate) [reply src/Infrastructure/Migrations/Version20230321102453.php:40]: Added below

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Kevin! :tada:

**Pull request merged**